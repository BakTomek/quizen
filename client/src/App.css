@import url('https://fonts.googleapis.com/css2?family=Schibsted+Grotesk:ital,wght@0,400..900;1,400..900&display=swap');

body{
    width: 100vw;
    height: 100vh;
    margin: 0;
}

#root{
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-direction: row;
    overflow-x: auto;
    overflow-y: hidden;
    font-family: "Schibsted Grotesk", sans-serif;
    font-optical-sizing: auto;
    font-weight: 400;
    font-style: normal;
}

/* width */
::-webkit-scrollbar {
    width: 10px;
  }
  
  /* Track */
  ::-webkit-scrollbar-track {
    background: transparent; 
  }
   
  /* Handle */
  ::-webkit-scrollbar-thumb {
    background: #537E99; 
    border-radius: 8px;
  }
  
  /* Handle on hover */
  ::-webkit-scrollbar-thumb:hover {
    background: #1B4863; 
  }