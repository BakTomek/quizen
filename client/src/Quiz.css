#QuizDiv{
    width: 100%;
    height: 99%;
    padding: 15px;
    overflow-y: scroll;
    background-color: #E9EFF2;
    display: flex;
    align-items: safe center;
    flex-direction: column;
    font-size: 18px;
    font-weight: 300;
    font-weight: 400;
    line-height: 26px;
    letter-spacing: 1%;
    color: #02090D;
}

#QuizDiv h1{
    color: #1B4863;
    font-size: 40px;
    line-height: 48px;
    letter-spacing: -1%;
    margin-bottom: 36px;
}

.QuizDescDiv{
    text-wrap: wrap;
    margin-top: 0;
    margin-bottom: 0;
}

.QuestionsAndAnswersDiv{
    margin-top: 54px;
    display: flex;
    justify-content: safe center;
    align-items: safe center;
    flex-direction: column;
    width: auto;
    min-width: 60%;
}

.QuestionsAndAnswersDiv h2{
    padding-bottom: 18px;
    margin-top: 54px;
    border-bottom: 2px solid #537E99;
    width: 80%;
    text-align: center;
}

.AnswersButtonsDiv{
    margin-top: 18px;
    display: flex;
    align-items: center;
    justify-content: space-around;
    min-width: 70%;
    max-width: 100%;
    width: auto;
    height: auto;
    flex-wrap: wrap;
}

.AnswersButtonsDiv button{
    background-color: #537E99;
    border: none;
    width: auto;
    min-width: 60px;
    height: auto;
    min-height: 30px;
    color: #E9EFF2;
    font-size: 16px;
    font-weight: 600;
    border-radius: 8px;
    border: 2px outset #537E99;
    padding: 12px;
    text-align: center;
    margin-bottom: 18px;
    margin-right: 10px;
    margin-left: 10px;
}

.AnswersButtonsDiv button:hover{
    background-color: #b19d50;
    transition: .2s;
    cursor: pointer;
    font-weight: 600;
    border: 2px inset #b19d50;
    transform: scale(1.1);
}