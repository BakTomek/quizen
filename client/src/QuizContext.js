import React, { createContext, useState } from 'react';

const QuizContext = createContext();

const QuizProvider = ({ children }) => {
  const [quizzes, setQuizzes] = useState([]);
  const [currentQuiz, setCurrentQuiz] = useState(null);
  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
  const [score, setScore] = useState(0);

  return (
    <QuizContext.Provider value={{
      quizzes,
      setQuizzes,
      currentQuiz,
      setCurrentQuiz,
      currentQuestionIndex,
      setCurrentQuestionIndex,
      score,
      setScore
    }}>
      {children}
    </QuizContext.Provider>
  );
};

export { QuizContext, QuizProvider };