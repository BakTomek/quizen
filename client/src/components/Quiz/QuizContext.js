import React, { createContext, useState } from 'react';

export const QuizContext = createContext();

export const QuizProvider = ({ children }) => {
  const [quizzes, setQuizzes] = useState([]);
  const [currentQuiz, setCurrentQuiz] = useState(null);
  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
  const [score, setScore] = useState(0);

  return (
    <QuizContext.Provider value={{ quizzes, setQuizzes, currentQuiz, setCurrentQuiz, currentQuestionIndex, setCurrentQuestionIndex, score, setScore }}>
      {children}
    </QuizContext.Provider>
  );
};
