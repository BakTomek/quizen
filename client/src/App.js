/*import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar/Navbar';
import MainPage from './components/MainPage/MainPage';
import Quizzes from './components/Quizzes/Quizzes';
import CreateQuiz from './components/CreateQuiz/CreateQuiz';
import Quiz from './components/Quiz/Quiz';
import { QuizProvider } from './components/Quiz/QuizContext';
import './App.css';

const App = () => {
  return (
    <QuizProvider>
      <Router>
        <Navbar />
        <Routes>
          <Route path="/" element={<MainPage />} />
          <Route path="/quizzes" element={<Quizzes />} />
          <Route path="/create-quiz" element={<CreateQuiz />} />
          <Route path="/create-quiz/:quizId" element={<CreateQuiz />} />
          <Route path="/quiz/:quizId" element={<Quiz />} />
        </Routes>
      </Router>
    </QuizProvider>
  );
};

export default App;*/
// App.js
import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar/Navbar';
import MainPage from './components/MainPage/MainPage';
import Quizzes from './components/Quizzes/Quizzes';
import CreateQuiz from './components/CreateQuiz/CreateQuiz';
import Quiz from './components/Quiz/Quiz';
import { QuizProvider } from './components/Quiz/QuizContext';
import './App.css';

const AppContent = () => (
  <>
    <Navbar />
    <Routes>
      <Route path="/" element={<MainPage />} />
      <Route path="/quizzes" element={<Quizzes />} />
      <Route path="/create-quiz" element={<CreateQuiz />} />
      <Route path="/create-quiz/:quizId" element={<CreateQuiz />} />
      <Route path="/quiz/:quizId" element={<Quiz />} />
    </Routes>
  </>
);

const App = () => (
  <QuizProvider>
    <Router>
      <AppContent />
    </Router>
  </QuizProvider>
);

export default App;
export { AppContent }; // Export for testing without Router
