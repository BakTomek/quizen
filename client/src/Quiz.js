import React, { useContext, useState, useEffect } from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import { QuizContext } from './QuizContext';
import './Quiz.css';
import back from './background_bruh_3.png';
import Swal from 'sweetalert2';

const divStyle = {
  backgroundImage: `url(${back})`,
  backgroundSize: 'cover',
  backgroundRepeat: 'no-repeat',
  backgroundPosition: 'center'
};

const Quiz = () => {
  const { quizId } = useParams();
  const { quizzes, setScore, score } = useContext(QuizContext);
  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
  const [currentQuiz, setCurrentQuiz] = useState(null);
  const navigate = useNavigate();

  useEffect(() => {
    const quiz = quizzes.find(q => q.quiz_id === parseInt(quizId));
    setCurrentQuiz(quiz);
  }, [quizId, quizzes]);

  if (!currentQuiz) {
    return <div>Loading...</div>;
  }

  const currentQuestion = currentQuiz.questions[currentQuestionIndex];

  const handleAnswerClick = (isCorrect) => {
    if (isCorrect) {
      setScore(score + 1);
    }

    if (currentQuestionIndex < currentQuiz.questions.length - 1) {
      setCurrentQuestionIndex(currentQuestionIndex + 1);
    } else {
      Swal.fire({
        title: 'Quiz completed!',
        text: `Your score is: ${score + (isCorrect ? 1 : 0)} / ${currentQuiz.questions.length}`,
        icon: 'success',
        confirmButtonText: 'OK'
      }).then(() => {
        navigate('/');
      });
    }
  };

  return (
    <div id='QuizDiv' style={divStyle}>
      <h1>{currentQuiz.title}</h1>
      <p className='QuizDescDiv'>{currentQuiz.description}</p>
      <div className='QuestionsAndAnswersDiv'>
        <h2>{currentQuestion.question_text}</h2>
        <div className='AnswersButtonsDiv'>
          {currentQuestion.answers.map((answer, index) => (
            <button key={index} onClick={() => handleAnswerClick(answer.is_correct)}>
              {answer.answer_text}
            </button>
          ))}
        </div>
      </div>
    </div>
  );
};

export default Quiz;