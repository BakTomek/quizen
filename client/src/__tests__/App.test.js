// App.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { QuizProvider } from '../components/Quiz/QuizContext';
import { AppContent } from '../App'; // Import AppContent for testing

// Mock Components
jest.mock('../components/Navbar/Navbar', () => () => <div>Mock Navbar</div>);
jest.mock('../components/MainPage/MainPage', () => () => <div>Mock Main Page</div>);
jest.mock('../components/Quizzes/Quizzes', () => () => <div>Mock Quizzes</div>);
jest.mock('../components/CreateQuiz/CreateQuiz', () => () => <div>Mock Create Quiz</div>);
jest.mock('../components/Quiz/Quiz', () => () => <div>Mock Quiz</div>);

describe('App', () => {
  test('renders Navbar', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.getByText('Mock Navbar')).toBeInTheDocument();
  });

  test('renders MainPage component for default route', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.getByText('Mock Main Page')).toBeInTheDocument();
  });

  test('renders Quizzes component for /quizzes route', () => {
    render(
      <MemoryRouter initialEntries={['/quizzes']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.getByText('Mock Quizzes')).toBeInTheDocument();
  });

  test('renders CreateQuiz component for /create-quiz route', () => {
    render(
      <MemoryRouter initialEntries={['/create-quiz']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.getByText('Mock Create Quiz')).toBeInTheDocument();
  });

  test('renders CreateQuiz component for /create-quiz/:quizId route', () => {
    render(
      <MemoryRouter initialEntries={['/create-quiz/1']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.getByText('Mock Create Quiz')).toBeInTheDocument();
  });

  test('renders Quiz component for /quiz/:quizId route', () => {
    render(
      <MemoryRouter initialEntries={['/quiz/1']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.getByText('Mock Quiz')).toBeInTheDocument();
  });

  test('renders NotFound component for unknown route', () => {
    render(
      <MemoryRouter initialEntries={['/unknown']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(screen.queryByText('Mock Main Page')).not.toBeInTheDocument();
    expect(screen.queryByText('Mock Quizzes')).not.toBeInTheDocument();
    expect(screen.queryByText('Mock Create Quiz')).not.toBeInTheDocument();
    expect(screen.queryByText('Mock Quiz')).not.toBeInTheDocument();
  });

  test('App component matches snapshot for / route', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('App component matches snapshot for /quizzes route', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/quizzes']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('App component matches snapshot for /create-quiz route', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/create-quiz']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('App component matches snapshot for /quiz/:quizId route', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/quiz/1']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('App component matches snapshot for unknown route', () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/unknown']}>
        <QuizProvider>
          <AppContent />
        </QuizProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
