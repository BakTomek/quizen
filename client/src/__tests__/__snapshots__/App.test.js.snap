// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component matches snapshot for / route 1`] = `
<DocumentFragment>
  <div>
    Mock Navbar
  </div>
  <div>
    Mock Main Page
  </div>
</DocumentFragment>
`;

exports[`App App component matches snapshot for /create-quiz route 1`] = `
<DocumentFragment>
  <div>
    Mock Navbar
  </div>
  <div>
    Mock Create Quiz
  </div>
</DocumentFragment>
`;

exports[`App App component matches snapshot for /quiz/:quizId route 1`] = `
<DocumentFragment>
  <div>
    Mock Navbar
  </div>
  <div>
    Mock Quiz
  </div>
</DocumentFragment>
`;

exports[`App App component matches snapshot for /quizzes route 1`] = `
<DocumentFragment>
  <div>
    Mock Navbar
  </div>
  <div>
    Mock Quizzes
  </div>
</DocumentFragment>
`;

exports[`App App component matches snapshot for unknown route 1`] = `
<DocumentFragment>
  <div>
    Mock Navbar
  </div>
</DocumentFragment>
`;
